test/integration/resource.ts

```diff
@@ -0,0 +1,41 @@
+import {expect} from 'chai';
+import {Ketting, NeverCache} from '../../src';
+
+describe('Using resources', async () => {
+  const ketting = new Ketting('http://localhost:3000/hal1.json');
+
+  it('concurrent get on the same resource should do 2 distinct HTTP calls if request headers are different', async () => {
+    ketting.cache = new NeverCache();
+
+    const resource = await ketting.follow('collection');
+
+    const [firstState, secondState] =
+      await Promise.all([
+        resource.get({headers: {'delay-in-ms': '2', 'foo': 'bar'}}),
+        resource.get({headers: {'foo': 'baz'}})
+      ]);
+
+    expect(firstState.timestamp).to.not.eql(secondState.timestamp);
```

Comment:
Instead, make the test server return `preference-applied` header based on the provided `prefer` header

Reply:
Done